Mathlib/RingTheory/Ideal/AssociatedPrime/Basic.lean

```diff
@@ -92,9 +103,54 @@ theorem exists_le_isAssociatedPrime_of_isNoetherianRing [H : IsNoetherianRing R]
   rwa [H₁.eq_of_not_lt (h₃ _ ⟨l.trans H₁, H₂, _, rfl⟩),
     mem_ker, toSpanSingleton_apply, smul_comm, smul_smul]
 
+/-- If `M → M'` is injective, then the set of associated primes of `M` is
+contained in that of `M'`. -/
+@[stacks 02M3 "first part"]
 theorem associatedPrimes.subset_of_injective (hf : Function.Injective f) :
     associatedPrimes R M ⊆ associatedPrimes R M' := fun _I h => h.map_of_injective f hf
 
+/-- If `0 → M → M' → M''` is an exact sequence, then the set of associated primes of `M'` is
+contained in the union of those of `M` and `M''`. -/
+@[stacks 02M3 "second part"]
+theorem associatedPrimes.subset_union_of_exact
+    {M'' : Type*} [AddCommGroup M''] [Module R M'']
+    (g : M' →ₗ[R] M'') (hf : Function.Injective f) (hfg : Function.Exact f g) :
+    associatedPrimes R M' ⊆ associatedPrimes R M ∪ associatedPrimes R M'' := by
+  rintro p ⟨_, x, hx⟩
+  by_cases h : ∃ a ∈ p.primeCompl, ∃ y : M, f y = a • x
+  · obtain ⟨a, ha, y, h⟩ := h
+    refine Or.inl ⟨‹_›, y, le_antisymm (fun b hb ↦ ?_) (fun b hb ↦ ?_)⟩
```

Comment:
Can you use `left` instead of `Or.inl`? It is more readable.

Reply:
Sure.